src/main/java/net/glowstone/block/blocktype/BlockType.java

```diff
@@ -163,6 +163,18 @@ public boolean canPlaceAt(GlowBlock block, BlockFace against) {
         return true;
     }
 
+    /**
+     * Check whether the block can be placed by a given player at a given location.
+     *
+     * @param player The player who placed the block.
+     * @param block The location the block is being placed at.
+     * @param against The face the block is being placed against.
+     * @return Whether the placement is valid.
+     */
+    public boolean canPlaceAt(GlowPlayer player, GlowBlock block, BlockFace against) {
```

Comment:
It isn't redundant. It's an overload method.

Reply:
Ah. Why not add the param to the existing method instead, changing all the Block subclass override method and keep the one with lesser args as the overload?

Reply:
I agree, it would make more sense to make all pathways leads to the longer method, just so the bed won't mess up in new places.

Reply:
Are you suggesting to remove the overload and just change the parameters of the original method?

Reply:
Change the parameters of the original method, but add an overload that has the original parameters, with a null player.

Reply:
OK, I think I understand what you mean now. The reason I didn't do it like that is because most classes extending BlockType don't need GlowPlayer to be passed to `canPlaceAt`. It would be a huge diff to change the original method and all occurrences, so I wanted to avoid that too.

Reply:
You wouldn't be changing all occurrences. You would just be making it so when something calls `canPlaceAt(GlowBlock, BlockFace)`, it returns the result of `canPlaceAt(GlowPlayer, GlowBlock, BlockFace)`.